packages/web3-rpc-providers/src/web3_provider_publicnode.ts

```diff
@@ -0,0 +1,112 @@
+/*
+This file is part of web3.js.
+
+web3.js is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+web3.js is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { EthExecutionAPI, Web3APISpec } from 'web3-types';
+import { HttpProviderOptions } from 'web3-providers-http';
+import { Network, SocketOptions, Transport } from './types.js';
+import { Web3ExternalProvider } from './web3_provider.js';
+
+const isValid = (str: string) => str !== undefined && str.trim().length > 0;
+
+const websocketExclusions = [
+	Network.DYMENSION_MAINNET,
+	Network.DYMENSION_TESTNET,
+	Network.KAVA_MAINNET,
+	Network.CRONOS_MAINNET,
+	// deprecated
+	Network.POLYGON_MAINNET,
+];
+
+export class PublicNodeProvider<
+	API extends Web3APISpec = EthExecutionAPI,
+> extends Web3ExternalProvider<API> {
+	// eslint-disable-next-line default-param-last
+	public constructor(
+		network: Network = Network.ETH_MAINNET,
+		transport: Transport = Transport.HTTPS,
+		host = '',
+		providerConfigOptions?: HttpProviderOptions | SocketOptions,
+	) {
+		super(network, transport, '', host, providerConfigOptions);
+	}
+	public static readonly networkHostMap: { [key: string]: string } = {
+		[Network.POLYGON_AMOY]: 'polygon-amoy-bor-rpc.publicnode.com',
+		[Network.DYMENSION_MAINNET]: 'dymension-evm-rpc.publicnode.com',
+		[Network.DYMENSION_TESTNET]: 'dymension-testnet-evm-rpc.publicnode.com',
+		[Network.BLAST_MAINNET]: 'blast-rpc.publicnode.com',
```

Comment:
```suggestion
		[Network.BLAST_MAINNET]: 'blast-rpc',
```
lets takeout multiple instances of `.publicnode.com` for all URLs as its same for all